### PR TITLE
Fix ModifyNodePopUp Input

### PR DESCRIPTION
**Pull request by dscatalan**

### CODE REVIEW

File: js/main-canvas.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ class MainCanvas extends Canvas {
         // Draw prereq indicators
         for (let node of this.nodes) {
             for(let prereq of node.prereqs) {
-                const found = this.findNodeByClassCode(prereq)
+                const found = this.createPrereqLink(prereq)
```
Comment (KosOrKosm) [inline js/main-canvas.js]: I prefer the original function name. This name implies the function creates something when it does not. It just finds a node in the list.

File: js/nodes.js [hunk 2]
```diff
@@ -17,7 +17,9 @@
     x = 0
     y = 0
 
+    classPrefixNumber = ""
```
Comment (KosOrKosm) [inline js/nodes.js:20]: The original node attributes I created are essentially placeholders. They really need to be replaced with the actual data the node will hold. For instance, there is no "classDescription", even though that will definitely be something we use. We will probably have a "classMajor" field as well. Currently, the major is being treated as the class's code, which is incorrect. Would you be interested in adding these attributes?

**Review by KosOrKosm — CHANGES_REQUESTED** [review]
Hey Daisy, I have noted a few things I would like to see changed. Also, I went ahead and merged this branch with the latest main -- you will notice there are a lot more JS files now, and some code got moved from main-canvas.js to handlers.js.

Could you address my comments and make sure everything is still working? We can definitely merge this after that.

File: js/main-canvas.js (hunk 1 above)
Comment (dscatalan) [reply js/main-canvas.js]: Okay I got everything to work again with the refactored code and I reverted the name back to findNodeByClassCode() no problem.

File: js/nodes.js (hunk 2 above)
Comment (dscatalan) [reply js/nodes.js:20]: Okay I have added classDescription. Still need to fix classMajor

Comment (dscatalan) [reply js/nodes.js:20]: Okay We have classMajor fixed now!

**Comment by KosOrKosm** [comment]
Looks great! Thanks for the work

**Pull request merged**